File: java/incometax/contribution/PagIBIGContribution.java
/** 
 * Pinoy-Sweldo a module for computing Income Tax in the Philippines.
 * :copyright: 2020-Present by Daniel De Castro.
 * :license: MIT, see LICENSE for more details.
*/

package incometax.contribution;

public class PagIBIGContribution extends Contribution{

    // Calculates the Pag IBIG monthly contribution depending
    // on the user monthly income.

    public PagIBIGContribution(double monthlySalary) throws Exception{
        super(monthlySalary);
        setModel("PAG-IBIG");
        setMonthlyContribution(
            compute(monthlySalary)
        );
    }

    public double compute(double monthlySalary){
        if ( monthlySalary <= 1500 )
            return monthlySalary * 0.01;
        else if ( monthlySalary > 1500 && monthlySalary < 5000 )
            return monthlySalary * 0.02;
        else if ( monthlySalary >= 5000 )
            return 100;
        return 0.0;
    }
}

File: python/incometax/contributions/pagibig_contribution.py
from .contribution import Contribution

class PagIBIGContribution(Contribution):
    """
    Calculates the Pag IBIG monthly contribution depending
    on the user monthly income.

    """

    def __init__(self, monthly_salary = 0):
        super().__init__(monthly_salary)
        self.model = 'PAG-IBIG'
        self.monthly_contribution = self.compute(monthly_salary)

    def compute(self, monthly_salary):
        """
        accepts the monthly_salary 
        and returns the monthly contribution

        """
        if monthly_salary <= 1500:
            return monthly_salary * 0.01
        elif monthly_salary > 1500 and monthly_salary < 5000:
            return monthly_salary * 0.02
        elif monthly_salary >= 5000:
            return 100
